app/admin/quests/dashboard/[questId]/page.tsx

```diff
@@ -439,176 +439,184 @@ export default function Page({ params }: QuestIdProps) {
+          await AdminService.createDomain({
+            quest_id: questId.current,
+            name: step.data.domain_name,
+            desc: step.data.domain_desc,
+          });
+        } else if (step.type === "Balance") {
+          await AdminService.createBalance({
+            quest_id: questId.current,
+            name: step.data.balance_name,
+            desc: step.data.balance_desc,
+            contracts: step.data.balance_contracts,
+            cta: step.data.balance_cta,
+            href: step.data.balance_href,
+          });
+        }
+      } catch (error) {
+        console.error(`Error adding task of type ${step.type}:`, error);
       }
-      if (step.type === "TwitterFw") {
-        await AdminService.createTwitterFw({
-          quest_id: questId.current,
-          name: step.data.twfw_name,
-          desc: step.data.twfw_desc,
-          username: step.data.twfw_username,
-        });
-      } else if (step.type === "TwitterRw") {
-        await AdminService.createTwitterRw({
-          quest_id: questId.current,
-          name: step.data.twrw_name,
-          desc: step.data.twrw_desc,
-          post_link: step.data.twrw_post_link,
-        });
-      } else if (step.type === "Discord") {
-        await AdminService.createDiscord({
-          quest_id: questId.current,
-          name: step.data.dc_name,
-          desc: step.data.dc_desc,
-          invite_link: step.data.dc_invite_link,
-          guild_id: step.data.dc_guild_id,
-        });
-      } else if (step.type === "Custom") {
-        await AdminService.createCustom({
-          quest_id: questId.current,
-          name: step.data.custom_name,
-          desc: step.data.custom_desc,
-          cta: step.data.custom_cta,
-          href: step.data.custom_href,
-          api: step.data.custom_api,
-        });
-      } else if (step.type === "Domain") {
-        await AdminService.createDomain({
-          quest_id: questId.current,
-          name: step.data.domain_name,
-          desc: step.data.domain_desc,
-        });
-      } else if (step.type === "Balance") {
-        await AdminService.createBalance({
-          quest_id: questId.current,
-          name: step.data.balance_name,
-          desc: step.data.balance_desc,
-          contracts: step.data.balance_contracts,
-          cta: step.data.balance_cta,
-          href: step.data.balance_href,
-        });
-      }
-    });
-
-    await Promise.all(taskPromises);
+    }
   }, []);
 
   const handleDeleteTasks = useCallback(async (removedTasks: StepMap[]) => {
-    const taskPromises = removedTasks.map(async (step) => {
-      await AdminService.deleteTask({
-        id: step.data.id,
-      });
-    });
-
-    await Promise.all(taskPromises);
+    for (const step of removedTasks) {
+      try {
+        await AdminService.deleteTask({
+          id: step.data.id,
+        });
+      } catch (error) {
+        console.error("Error deleting task:", error);
+      }
+    }
   }, []);
```

Comment:
This part is not needed, only the handleAddTasks needs the for loop

Reply:
Okay